### PR TITLE
[fsdp] fix: optimize model updating in rollout mode in LoRA training with FSDP

### PR DESCRIPTION
**Pull request by Zhikaiiii**
### What does this PR do?

Lora training is much slower than full training under the same experiment setup. Just as https://github.com/volcengine/verl/issues/4033, https://github.com/volcengine/verl/issues/4085, https://github.com/volcengine/verl/issues/4055.

From the ray timeline and tensorboard, it seems that the main time discrepancy does not occur during the `generate sequence`, but rather during the switching of `rollout_mode`.

tensorboard:
image
image

full:
image

lora:
image

While dived into the code, I found that in version 0.6.1, when the vLLM version is greater than 0.8.5, the default SLEEP_LEVEL is changed to 2, causing the base_model to be reloaded every time rollout_mode is called, which is time-consuming

```Python
elif vs.parse(package_version) >= vs.parse("0.7.0"):
    vllm_version = package_version
    if vs.parse(package_version) >= vs.parse("0.8.5"):
        VLLM_SLEEP_LEVEL = 2
    from vllm import LLM
    from vllm.distributed import parallel_state
```

In `rollout_mode`, when using full training,  `fsdp_module.state_dict()` can be used directly. However, in the LoRA case, the `collect_lora_param` function first summons the parameters and then moves them to the CPU. I think this is time-consuming and unnecessary. 
https://github.com/volcengine/verl/blob/v0.6.1/verl/workers/fsdp_workers.py#L672
https://github.com/volcengine/verl/blob/v0.6.1/verl/utils/fsdp_utils.py#L629

So I modify the lora weight updating schema in this pr.

### Test

Run experiments in both fsdp and fsdp2 with Qwen2.5-7b in gsm8k datasets. The results are as belowed：
- qwen25_7b_lora_fsdp2_async： fsdp2 + lora + fix-code
- qwen25_7b_lora_fsdp_async: fsdp + lora + fix-code
- qwen25_7b_lora_fsdp2_async_old: fsdp2 + lora + current-code
- qwen25_7b_lora_fsdp_async_old: fsdp + lora + current-code
- qwen25_7b_full_fsdp2_async: fsdp2 + full + current-code, for comparsion.

The reward function's curve aligns with expectations, while the training time per step has been significantly reduced.

training_reward:
image
val_reward:
image

timing/generate_sequence:
image

timing/gen (with time of rollout mode)
image

timing/step
image

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [x] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)

### CODE REVIEW

**Comment by Zhikaiiii** [comment]
@PeterSH6 @vermouth1992 @wuxibin89 hello, the ci test have passed, could you please take a look at this pr : )
